swaybar/ipc.c

```diff
@@ -305,22 +305,19 @@ bool ipc_get_workspaces(struct swaybar *bar) {
 				ws->num = json_object_get_int(num);
 				ws->name = strdup(json_object_get_string(name));
 				ws->label = strdup(ws->name);
-				char *ws_dup = strdup(ws->name);
-				char *stripped_name = strtok(ws_dup, ":");
-				size_t num_len = strlen(stripped_name) + 1;
-				// ws->num will be 0 if workspace name doesn't begin with int.
-				if (stripped_name != NULL && 
-						strlen(ws->name) > num_len && ws->num != -1) {
-					if(bar->config->strip_workspace_name) {
+				size_t len_offset = ws->name[1] == ':' ? 2 : 1;
+				// ws->num will be -1 if workspace name doesn't begin with int.
```

Comment:
This code makes the assumption that the digit will always be a single character (0-9). Not ideal, but existing code already does this.